### ISSUE TITLE
[mu4e bug] Undefined symbol docid on 1.8

### ISSUE THREAD

**Issue by drtutut** [djcb/mu#2432]
The context is a migration from mu 1.4 to mu 1.8.14

**Describe the bug**

Starting mu4e is ok.  trying to jump to a headers view systematically generates an error:
`error in process filter: Symbol’s function definition is void: docid`

**How to Reproduce**

F1 to start mu4e, then j ou b shortcut to jump to a maildir or a view.

**Environment**
Ubuntu 20.04, emacs 26.3, mu/mu4e 1.8.14

I do not see any suspect setting in my configuration, but I will try to deactivate some settings.

**Checklist**
- [X] you are running either an 1.8.x release or the latest 1.9.x development release
- [X] you can reproduce the problem without 3rd party extensions (including Doom/Evil etc.)
- [X] you have read all of the above

**Comment by drtutut** [comment]
Additional remark : if the view contains to messages, the error does not occur.

**Comment by djcb** [comment]
This report needs a bit more info:

1) Please provide the emacs backtrace (i.e., `M-x toggle-debug-on-error`, reproduce and post the results here
2) Are you sure you have cleared up and remnants from the older version (such as `.elc` files)?
3) Does it happen with the most minimal setup (`emacs -Q` and only load the most basic mu4e config)

**Comment by drtutut** [comment]
Thank you for you answer.

So I :

1. checked `mu4e-*.elc` files hanging around: there isn't any.
2. started emacs with `emacs -Q`
3. loaded only this config:
```
(add-to-list 'load-path "/usr/local/share/emacs/site-lisp/mu4e")
(require 'mu4e)
(require 'smtpmail)

;; folders
(setq
 mu4e-maildir       "my-mail-dir"   ;; top-level Maildir 

 mu4e-sent-folder   "/sent"       ;; folder for sent messages
 mu4e-drafts-folder "/drafts"     ;; unfinished messages
 mu4e-trash-folder  "/trash"      ;; trashed messages
 mu4e-refile-folder "/archive")   ;; saved messages

(setq
 mu4e-get-mail-command "my-offlineimap-command"   
 mu4e-update-interval 300
 ;; smtp
 message-send-mail-function 'smtpmail-send-it
 smtpmail-stream-type 'starttls

 ;; attachment dir
 mu4e-attachment-dir  "~/Téléchargements"
 
 ;; insert signature
 mu4e-compose-signature-auto-include 't)
```

4. toggled `toggle-debug-on-error`
5. launched mu4e with command `mu4e`
6. try to use the `bt` shortcut, which leads to the following error
```
Debugger entered--Lisp error: (void-function docid)
  docid(2472)
  #f(compiled-function (msg) #<bytecode 0x140003d>)((:path "/home/eric/Mail/vs/vs/cur/1676207226_0.9904.leventikos,U=40644,FMD5=7e33429f656f1e6e9d79b29c3f82c57e:2,S" :size 7655 :list-post ((:email "reply+ABKIFYYZ7TEFAU27KMD5BT6B6YLNXEVBNHHF4P4Q3Y@reply.github.com")) :reply-to ((:name "djcb/mu" :email "reply+ABKIFYYZ7TEFAU27KMD5BT6B6YLNXEVBNHHF4P4Q3Y@reply.github.com")) :cc ((:name "Eric Würbel" :email "eric@vents-sauvages.fr") (:name "Author" :email "author@noreply.github.com")) :changed (25576 64879 0) :date (25576 58459 0) :flags (seen list) :from ((:name "Dirk-Jan C. Binnema" :email "notifications@github.com")) :list "mu.djcb.github.com" :message-id "djcb/mu/issues/2432/1427028740@github.com" :priority low :references ("djcb/mu/issues/2432@github.com") :subject "Re: [djcb/mu] [mu4e bug] (Issue #2432)" :to ((:name "djcb/mu" :email "mu@noreply.github.com")) :maildir "/vs/vs" :docid 2472 :meta (:path "0:z" :level 0 :date "n63e8e45b" :date-tstamp (0 0 0) :root t :thread-subject t)))
  mapc(#f(compiled-function (msg) #<bytecode 0x140003d>) ((:path "/home/eric/Mail/vs/vs/cur/1676207226_0.9904.leventikos,U=40644,FMD5=7e33429f656f1e6e9d79b29c3f82c57e:2,S" :size 7655 :list-post ((:email "reply+ABKIFYYZ7TEFAU27KMD5BT6B6YLNXEVBNHHF4P4Q3Y@reply.github.com")) :reply-to ((:name "djcb/mu" :email "reply+ABKIFYYZ7TEFAU27KMD5BT6B6YLNXEVBNHHF4P4Q3Y@reply.github.com")) :cc ((:name "Eric Würbel" :email "eric@vents-sauvages.fr") (:name "Author" :email "author@noreply.github.com")) :changed (25576 64879 0) :date (25576 58459 0) :flags (seen list) :from ((:name "Dirk-Jan C. Binnema" :email "notifications@github.com")) :list "mu.djcb.github.com" :message-id "djcb/mu/issues/2432/1427028740@github.com" :priority low :references ("djcb/mu/issues/2432@github.com") :subject "Re: [djcb/mu] [mu4e bug] (Issue #2432)" :to ((:name "djcb/mu" :email "mu@noreply.github.com")) :maildir "/vs/vs" :docid 2472 :meta (:path "0:z" :level 0 :date "n63e8e45b" :date-tstamp (0 0 0) :root t :thread-subject t))))
  seq-do(#f(compiled-function (msg) #<bytecode 0x140003d>) ((:path "/home/eric/Mail/vs/vs/cur/1676207226_0.9904.leventikos,U=40644,FMD5=7e33429f656f1e6e9d79b29c3f82c57e:2,S" :size 7655 :list-post ((:email "reply+ABKIFYYZ7TEFAU27KMD5BT6B6YLNXEVBNHHF4P4Q3Y@reply.github.com")) :reply-to ((:name "djcb/mu" :email "reply+ABKIFYYZ7TEFAU27KMD5BT6B6YLNXEVBNHHF4P4Q3Y@reply.github.com")) :cc ((:name "Eric Würbel" :email "eric@vents-sauvages.fr") (:name "Author" :email "author@noreply.github.com")) :changed (25576 64879 0) :date (25576 58459 0) :flags (seen list) :from ((:name "Dirk-Jan C. Binnema" :email "notifications@github.com")) :list "mu.djcb.github.com" :message-id "djcb/mu/issues/2432/1427028740@github.com" :priority low :references ("djcb/mu/issues/2432@github.com") :subject "Re: [djcb/mu] [mu4e bug] (Issue #2432)" :to ((:name "djcb/mu" :email "mu@noreply.github.com")) :maildir "/vs/vs" :docid 2472 :meta (:path "0:z" :level 0 :date "n63e8e45b" :date-tstamp (0 0 0) :root t :thread-subject t))))
  mu4e~headers-append-handler(((:path "/home/eric/Mail/vs/vs/cur/1676207226_0.9904.leventikos,U=40644,FMD5=7e33429f656f1e6e9d79b29c3f82c57e:2,S" :size 7655 :list-post ((:email "reply+ABKIFYYZ7TEFAU27KMD5BT6B6YLNXEVBNHHF4P4Q3Y@reply.github.com")) :reply-to ((:name "djcb/mu" :email "reply+ABKIFYYZ7TEFAU27KMD5BT6B6YLNXEVBNHHF4P4Q3Y@reply.github.com")) :cc ((:name "Eric Würbel" :email "eric@vents-sauvages.fr") (:name "Author" :email "author@noreply.github.com")) :changed (25576 64879 0) :date (25576 58459 0) :flags (seen list) :from ((:name "Dirk-Jan C. Binnema" :email "notifications@github.com")) :list "mu.djcb.github.com" :message-id "djcb/mu/issues/2432/1427028740@github.com" :priority low :references ("djcb/mu/issues/2432@github.com") :subject "Re: [djcb/mu] [mu4e bug] (Issue #2432)" :to ((:name "djcb/mu" :email "mu@noreply.github.com")) :maildir "/vs/vs" :docid 2472 :meta (:path "0:z" :level 0 :date "n63e8e45b" :date-tstamp (0 0 0) :root t :thread-subject t))))
  mu4e--server-filter(#<process  *mu4e-server*> "\376b\377(:erase t)\n\3763bf\377(:headers ((:path \"/home/eric/Mail/vs/vs/cur/1676207226_0.9904.leventikos,U=40644,FMD5=7e33429f656f1e6e9d79b29c3f82c57e:2,S\" :size 7655 :list-post ((:email \"reply+ABKIFYYZ7TEFAU27KMD5BT6B6YLNXEVBNHHF4P4Q3Y@reply.github.com\")) :reply-to ((:name \"djcb/mu\" :email \"reply+ABKIFYYZ7TEFAU27KMD5BT6B6YLNXEVBNHHF4P4Q3Y@reply.github.com\")) :cc ((:name \"Eric W\303\274rbel\" :email \"eric@vents-sauvages.fr\") (:name \"Author\" :email \"author@noreply.github.com\")) :changed (25576 64879 0) :date (25576 58459 0) :flags (seen list) :from ((:name \"Dirk-Jan C. Binnema\" :email \"notifications@github.com\")) :list \"mu.djcb.github.com\" :message-id \"djcb/mu/issues/2432/1427028740@github.com\" :priority low :references (\"djcb/mu/issues/2432@github.com\") :subject \"Re: [djcb/mu] [mu4e bug] (Issue #2432)\" :to ((:name \"djcb/mu\" :email \"mu@noreply.github.com\")) :maildir \"/vs/vs\" :docid 2472 :meta (:path \"0:z\" :level 0 :date \"n63e8e45b\" :date-tstamp (0 0 0) :root t :thread-subject t))\n))\n\376b\377(:found 1)\n;; mu> ")

```

**Comment by drtutut** [comment]
FYI, I performed a clean uninstall of 1.8.4 and installed a 1.6.11 instead, and it runs without problems (I need my email !). 

But if you need more data on this problem with 1.8.4, I can reinstall it.

**Comment by djcb** [comment]
Ah... if it works with 1.9.x that's good news at least! 

Thanks for the debugging  I suppose you mean 1.8.14, not 1.8.4? Anyway, if you have time, getting a lisp-level backtrace from uncompiled code would be interesting.

**Comment by drtutut** [comment]
Yes, sorry, I meant 1.8.14.
(and you probably mean 1.6.x, not 1.9.x ;-)

Ok, I will redo the test with uncompiled code.

**Comment by djcb** [comment]
Closing, since we've just released 1.10. Feel to reopen if you still see this (with the backtraces of uncompiled code). Thanks!
